src/context/ChartContextProvider.tsx

```diff
@@ -155,16 +153,64 @@ function useTidyDataToChartContext(
       };
 
       // The Y values object also passes through the user specified series properies from dataSelection
-      const yValues = {
+      let yValues = {
         name: series.displayName,
         color: series.color,
         dashStyle: series.dashStyle,
+        intervalType: "---",
         values: yArray,
       };
 
-      // Push the X and Y values objects to the chart data
-      newChartData.xValues.push(xValues);
-      newChartData.yValues.push(yValues);
+      if (
+        series.intervalType !== "---" &&
+        series.lowerBoundSeries !== "" &&
+        series.upperBoundSeries !== ""
+      ) {
+        // Extract the X and Y values from columns into arrays
+        const xLowerArray = arrayColumn(currentSeries, category);
+        const yLowerArray = arrayColumn(currentSeries, series.lowerBoundSeries);
+
+        const xUpperArray = arrayColumn(currentSeries, category).reverse();
+        const yUpperArray = arrayColumn(
+          currentSeries,
+          series.upperBoundSeries,
+        ).reverse();
+
+        // Create X and Y values objects for the current series
+        const xConfidenceValues = {
+          name: category,
+          values: xLowerArray.concat(xUpperArray),
+        };
+
+        // The Y values object also passes through the user specified series properies from dataSelection
+        const newColor = series.intervalColor
+          .replace(")", ", 0.2)")
+          .replace("rgb", "rgba");
+        const yConfidenceValues = {
+          name: "Confidence Intervals",
+          color: newColor,
+          dashStyle: series.dashStyle,
+          intervalType:
+            series.intervalType === "error bars"
```

Comment:
This should probably at least be sentence cased (`Error bars`)

Reply:
Changed to Error bars and Intervals